include/swift/AST/DiagnosticsSema.def

```diff
@@ -3043,9 +3043,12 @@ ERROR(transpose_attr_overload_not_found,none,
 ERROR(transpose_attr_cannot_use_named_wrt_params,none,
       "cannot use named 'wrt' parameters in '@transpose(of:)' attribute, found "
       "%0", (Identifier))
-ERROR(transpose_attr_result_value_not_differentiable,none,
-      "'@transpose(of:)' attribute requires original function result %0 to "
-      "conform to 'Differentiable'", (Type))
+ERROR(transpose_func_needs_static,none,
```

Comment:
```suggestion
ERROR(transpose_func_wrt_self_must_be_static,none,
```

How about this message:
```
the transpose of an instance method must be a 'static' method in the same type when 'self' is a linearity parameter
```

Some notes:
* [Linearity parameter](https://github.com/apple/swift/pull/28925/files#diff-387578f048b82b8cc9df640a5fff9215R1653) is the new standard name for a `@transpose` attribute "wrt" parameter
* Keywords like `'static'` should be enclosed in single quotes
* Diagnostic messages do not end with a period

With this wording, this error can subsume `transpose_func_self_static_types_not_match`.